sources/models/scalar-derive/src/lib.rs

```diff
@@ -181,15 +218,24 @@ use syn::{parse_macro_input, Data, DataStruct, DeriveInput, Fields};
 pub fn scalar(input: TokenStream) -> TokenStream {
     // Parse the input tokens.
     let derive_input = parse_macro_input!(input as DeriveInput);
+    let name = (&derive_input.ident).to_string();
     let settings = RawSettings::from_attributes(derive_input.attrs.as_slice())
         .expect("Unable to parse `scalar` macro arguments");
 
-    // Further parse the input.
-    let struct_info = StructInfo::new(&derive_input, settings);
-
-    // Write impls.
+    // We will write syntax as tokens to this token stream.
     let mut ast2 = TokenStream2::new();
-    struct_info.write_impls(&mut ast2);
+
+    match &derive_input.data {
+        Data::Struct(s) => {
+            // Further parse the input.
+            let struct_info = StructInfo::new(&name, &s, settings);
+            struct_info.write_impls(&mut ast2);
+        }
+        Data::Enum(_) => write_string_impls_for_enum(&name, &mut ast2),
+        Data::Union(_) => panic!("A Scalar cannot be an union, it must be a struct"),
```

Comment:
```suggestion
        Data::Union(_) => panic!("A Scalar cannot be an union, it must be a struct or enum"),
```
Is this more accurate?